Panic in the distributor

```
panic: runtime error: slice bounds out of range

goroutine 225 [running]:
panic(0xaf7560, 0xc4200160a0)
    /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/weaveworks/cortex/chunk.(*AWSStore).lookupChunksForMatcher(0xc42026c640, 0x7fe601c4b4d0, 0xc420484fc0, 0xc42014de67, 0x1, 0x7ffdacdb3920, 0xc, 0xc4206e8180, 0x6, 0xc4206c26e0, ...)
    /go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:572 +0xff7
github.com/weaveworks/cortex/chunk.(*AWSStore).lookupChunksFor.func1(0xc42026c640, 0x7fe601c4b4d0, 0xc420484fc0, 0xc42014de67, 0x1, 0x7ffdacdb3920, 0xc, 0xc4206e8180, 0x6, 0xc4206c26e0, ...)
    /go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:505 +0xcf
created by github.com/weaveworks/cortex/chunk.(*AWSStore).lookupChunksFor
    /go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:511 +0x17b
panic: runtime error: slice bounds out of range
```

--- a/chunk/chunk_store.go
+++ b/chunk/chunk_store.go
@@ -486,7 +486,11 @@ func (c *AWSStore) lookupChunks(ctx context.Context, userID string, from, throug
 }
 
 func next(s string) string {
-	// TODO deal with overflows
+	if len(s) == 0 {
+		return "\x01" // Result cannot container null characters, so use 1 instead.
+	}
+
+	// TODO deal with overflows, null characters etc
 	l := len(s)
 	result := s[:l-1] + string(s[l-1]+1)
 	return result

--- a/chunk/chunk_store_test.go
+++ b/chunk/chunk_store_test.go
@@ -136,6 +136,7 @@ func TestChunkStore(t *testing.T) {
 	nameMatcher := mustNewLabelMatcher(metric.Equal, model.MetricNameLabel, "foo")
 
 	test("Just name label", []Chunk{chunk1, chunk2}, nameMatcher)
+	test("Empty matcher", nil, nameMatcher, mustNewLabelMatcher(metric.Equal, "bar", ""))
 	test("Equal", []Chunk{chunk1}, nameMatcher, mustNewLabelMatcher(metric.Equal, "bar", "baz"))
 	test("Not equal", []Chunk{chunk2}, nameMatcher, mustNewLabelMatcher(metric.NotEqual, "bar", "baz"))
 	test("Regex match", []Chunk{chunk1, chunk2}, nameMatcher, mustNewLabelMatcher(metric.RegexMatch, "bar", "beep|baz"))

